packages/types/lib/IQueryContext.ts

```diff
@@ -24,7 +23,6 @@ export interface IQueryContextCommon {
   // Types of these entries should be aligned with contextKeyShortcuts in ActorInitQueryBase
   // and Keys in @comunica/context-entries
 
-  source?: IDataSource;
```

Comment:
Why can't we just keep this instead of `sourceBinding`?

Reply:
Mainly because IDataSource means it can be 4 different types. This will make the mergeHandler function more complex as making a Set of objects won't work properly (as far as I know). It is possible, but will require more type checks and might be slightly more expensive.

Reply:
Sets of objects should work fine AFAIK, as Sets will just hash the objects.

The problem with removing that source key is that it will break many other things that make use of that key. So I'd recommend keeping it as-is. And if we really need a different type, then I'd add a different key.

Reply:
I think it's the same 'problem' as with the keys, if you get two objects describing the same data source, then taking the set of them wouldn't deduplicate them. Unless you know for sure that if two bindings have the same data source object in content, they also reference the same object.

In any case, I don't remember why I removed the source entry in the context. So that should be fixed for sure.

Reply:
> I think it's the same 'problem' as with the keys, if you get two objects describing the same data source, then taking the set of them wouldn't deduplicate them.

Indeed, it's the same as with the keys. Sources will be equal by identity. So equal sources refer to the same objects, so Sets will dedupe them.